src/azure-cli-core/azure/cli/core/auth/util.py

```diff
@@ -43,11 +43,8 @@ def aad_error_handler(error, **kwargs):
         recommendation = PASSWORD_CERTIFICATE_WARNING
     else:
         login_command = _generate_login_command(**kwargs)
-        recommendation = (
-            # Cloud Shell uses IMDS-like interface for implicit login. If getting token/cert failed,
-            # we let the user explicitly log in to AAD with MSAL.
-            "Please explicitly log in with:\n{}" if error.get('error') == 'broker_error'
```

Comment:
`az account get-access-token --scope non-existing` triggers a `{'error': 'AudienceNotSupported', 'error_description': 'Audience non-existing is not a supported MSI token audience.'}`.

`error.get('error') == 'broker_error'` cannot handle that either.

Reply:
> if this error handling snippet in Azure CLI was meant to pinpoint that corner case

Yes. The error handing is indeed to pinpoint Cloud Shell's potential "deceivingly successful token response containing a mismatching token type" situation. 

Do you suggest me to change it to `if in_cloud_console() and error.get('error') == 'broker_error` so that we can still handle this error?